waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/metrics/MonitoringConfiguration.java

```diff
@@ -15,102 +15,86 @@
  */
 package com.hotels.bdp.waggledance.metrics;
 
-import java.io.Closeable;
-import java.net.InetSocketAddress;
-import java.util.HashSet;
-import java.util.Set;
 import java.util.concurrent.TimeUnit;
 
-import javax.annotation.PostConstruct;
-import javax.annotation.PreDestroy;
-
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 
-import com.codahale.metrics.JmxReporter;
-import com.codahale.metrics.MetricFilter;
-import com.codahale.metrics.MetricRegistry;
-import com.codahale.metrics.ScheduledReporter;
-import com.codahale.metrics.graphite.Graphite;
-import com.codahale.metrics.graphite.GraphiteReporter;
-import com.codahale.metrics.jvm.GarbageCollectorMetricSet;
-import com.codahale.metrics.jvm.MemoryUsageGaugeSet;
-import com.codahale.metrics.jvm.ThreadStatesGaugeSet;
+import io.micrometer.core.instrument.Clock;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.config.NamingConvention;
+import io.micrometer.core.instrument.simple.SimpleMeterRegistry;
+import io.micrometer.core.instrument.util.HierarchicalNameMapper;
+import io.micrometer.graphite.GraphiteConfig;
+import io.micrometer.graphite.GraphiteMeterRegistry;
+import io.micrometer.graphite.GraphiteProtocol;
+import io.micrometer.jmx.JmxConfig;
+import io.micrometer.jmx.JmxMeterRegistry;
 
 import com.hotels.bdp.waggledance.conf.GraphiteConfiguration;
 
 @Configuration
 public class MonitoringConfiguration {
-  private static final Logger LOG = LoggerFactory.getLogger(MonitoringConfiguration.class);
 
-  private final Set<Closeable> reporters = new HashSet<>();
-  private @Autowired MetricRegistry metricRegistry;
-  private @Autowired GraphiteConfiguration graphiteConfiguration;
+  @Bean
+  public MeterRegistry meterRegistry(GraphiteConfiguration graphiteConfiguration) {
```

Comment:
As discussed, maybe we can remove this entirely and just pass everything in on the CLI like seems to be the new standard in Spring boot. If everything here works "as is" then perhaps a good approach would be to merge all of this and then have a separate PR where we remove our own graphite configuration from the YAML (and code) and move to Spring Boot Micrometer instead.

Reply:
hang on that would break everything
can't we be a bit smarter about this? e.g. use our own legacy config or cli micrometer overrides going forward? At least for a while where we deprecate stuff and let give users a change to migrate (we'll need to update readme etc... as well).

Reply:
Yeah. Once I have it all working in this manner and its deployed. I can look at options to move to the Standard Spring Boot convention and will take care Patrick's concern then.